mmdet/models/roi_heads/roi_extractors/groie.py

```diff
@@ -0,0 +1,163 @@
+        return x
+
+
+def SequentialModule(modules):
+    """Apply a sequential module."""
+    nn_modules = []
+    for mc in modules:
+        type_ = mc.pop('type')
+        nn_modules.append(load_processing_model(type_, mc))
+    return nn.Sequential(*nn_modules)
+
+
+# Models supported by GRoIE
+models = {
+    'NonLocal2D': NonLocal2D,
+    'GeneralizedAttention': GeneralizedAttention,
+    'ConvModule': ConvModule,
+    'Sequential': SequentialModule,
+    'ReLU': nn.ReLU,
+}
+
+
+def load_processing_model(model_type, config):
+    """Load a specific module."""
+    return models.get(model_type, NoProcess)(**config)
+
+
+@ROI_EXTRACTORS.register_module
+class SumGenericRoiExtractor(SingleRoIExtractor):
+    """Extract RoI features from all summed feature maps levels.
+
+    Args:
+        pre_cfg (dict): Specify pre-processing modules.
+        post_cfg (dict): Specify post-processing modules.
+        ... (see SingleRoIExtractor for other parameters)
+    """
+
+    def __init__(self, **kwargs):
+        kwargs = deepcopy(kwargs)
+
+        post_cfg = kwargs.pop('post_cfg')
+        pre_cfg = kwargs.pop('pre_cfg')
+
+        pre_type = pre_cfg.pop('type')
+        post_type = post_cfg.pop('type')
+
+        super(SumGenericRoiExtractor, self).__init__(**kwargs)
+
+        # build pre/post processing modules
+        self.post_module = load_processing_model(post_type, post_cfg)
+        self.pre_module = load_processing_model(pre_type, pre_cfg)
+        self.relu = nn.ReLU(inplace=False)
+
+    @force_fp32(apply_to=('feats', ), out_fp16=True)
+    def forward(self, feats, rois, roi_scale_factor=None):
+        if len(feats) == 1:
+            return self.roi_layers[0](feats[0], rois)
+
+        out_size = self.roi_layers[0].out_size
+        num_levels = len(feats)
+        roi_feats = feats[0].new_zeros(
+            rois.size(0), self.out_channels, *out_size)
+
+        if roi_scale_factor is not None:
+            rois = self.roi_rescale(rois, roi_scale_factor)
+
+        for i in range(num_levels):
+            # apply pre-processing to a RoI extracted from each layer
+            rois_ = rois
+            roi_feats_t = self.roi_layers[i](feats[i], rois_)
+            roi_feats_t = self.pre_module(roi_feats_t)
+            roi_feats_t = self.relu(roi_feats_t)
+            # and sum them all
+            roi_feats += roi_feats_t
+
+        # apply post-processing before return the result
+        x = self.post_module(roi_feats)
+        return x
+
+
+@ROI_EXTRACTORS.register_module
+class ConcatGenericRoiExtractor(SingleRoIExtractor):
+    """Extract RoI features from all concatenated feature maps levels.
+
+    Args:
+        pre_cfg (dict): Specify pre-processing modules.
+        reduce_cfg (dict): Specify reduce function before post-processing.
+        post_cfg (dict): Specify post-processing modules.
+        ... (see SingleRoIExtractor for other parameters)
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
Following the MMDetection's code-style, it might be more consistent to put new arguments here, like 
```python
def __init__(self, post_cfg,  pre_cfg, reduce_cfg, **kwargs):
```
This can also make the API documentation more clear.

Reply:
done. thanks :smile: